AzureADSample/AzureADSample/View/MainPage.xaml.cs

```diff
@@ -0,0 +1,71 @@
+﻿using AzureADSample.Model;
+using AzureADSample.Service;
+using AzureADSample.Views;
+using Microsoft.Identity.Client;
+
+namespace AzureADSample;
+
+public partial class MainPage : ContentPage
+{
+    AuthService authService;
+    User user;
+    public MainPage()
+    {
+        InitializeComponent();
+        authService = new AuthService();
+        user = new User();
+    }
+
+    // If an existing account already exist
+    // then direct the user to the settigns page
+    protected override async void OnAppearing()
+    {
+        try
+        {
+            var accounts = await AuthService.authenticationClient.GetAccountsAsync();
+            AuthenticationResult result;
+
+            if (accounts.Count() >= 1)
+            {
+                result = await AuthService.authenticationClient
+                   .AcquireTokenSilent(Constants.Scopes, accounts.FirstOrDefault())
+                   .ExecuteAsync();
+
+                await SecureStorage.SetAsync("Token", result?.IdToken); // store token securely for later use
+                authService.GetUserClaims(result, user);
+                await DisplayAlert("Sucessful Logged in", "Existing account exist", "Ok");
```

Comment:
I was also a bit confused by this flow for navigating back. 

https://user-images.githubusercontent.com/21988533/181306057-4cfadcc1-32f9-4bac-95c6-2b34076cbc0e.mp4

One way you might consider approaching this is:
*Changing the second page to not have a back button, since there's no need to log in once you're already logged in
*Making the log out button bring you back to the first page

Let me know what you think!

Reply:
Fixed it and @gal42 will take the task signing out and redirecting back to the home page.

Reply:
Thanks for the clarification!

Instead of having it automatically log in if the account already exists - one idea is to have the pop-up give users the option. As you mention, it can be based on preference. So the pop-up can look something like:

> **Existing account detected**
> 
> Would you like to log back in as {{User}}?
> 
> Yes / No

And if they indicate No, it can just close the pop-up so users can log in as a new user from the first page

Right now, there is no way for the user to get back to the first page even when they try to log out - the infinite loop remains:

https://user-images.githubusercontent.com/21988533/182162028-63768ddc-6b2f-4cfb-8bc0-652e5833972a.mp4

Looking forward to seeing what you / @gal42 decide!
.

Reply:
Thanks for the suggestion I've added the user to be able to pick whether they want to continue with the current existing account or not.

Reply:
Awesome! Thanks for the quick update; it looks great